src/stdplus/clone.hpp

```diff
@@ -0,0 +1,18 @@
+
+#pragma once
+
+namespace stdplus {
+
+template <typename Base, typename Derived>
+class cloneable : public Base
+{
+	public:
+		using Base::Base;
+
+		virtual Base *clone() const {
```

Comment:
Depends.

If we use it to introduce later other inherited types (e.g. class foo and class bar) in place of Base template param, then it will be similar to:

error: virtual function 'clone' has a different return type ('unique_ptr<c_cable_udp_addr>') than the function it overrides (which has return type 'unique_ptr<c_cable_base_addr>

How ever usually we use it so that the Base is the same, in such case we could return same.
So we could use unique_ptr I suppose.
This code is not used after all.

Makring that it can be unique_ptr , and disabling this code.